File: server/src/main/java/auxtripper/server/main/models/Playlist.java
package auxtripper.server.main.models;

import java.util.List;

import lombok.Data;

@Data
public class Playlist {

    private List<String> genres;
    private List<SpotifyTrack> tracks;
    private List<SpotifyArtist> artists;
    private SpotifyVibe vibe;
    private PlaylistRequest playlistRequest;
    private String playlistId;
    
}

File: client/src/app/models/playlist.ts
import { Artist, Track, Vibe } from "./spotify-models"

export interface Playlist {
    playlistId: string
    playlistRequest: PlaylistRequest
    artists: Artist[]
    tracks: Track[]
    genres: string[]
    isCreated: boolean
    createdOn: Date
    vibe: Vibe
}

export interface PlaylistRequest {
    id: string
    title: string
    isPublic: boolean
    isCollaborative: boolean
    description: string
    songs: string[]
    imageData: string
}

export interface ModifyPlaylistRequest {
    dbPlaylistId: number
    playlistId: string
    title: string
    isPublic: boolean
    isCollaborative: boolean
    description: string
    songs: string[]
    tracks: Track[]
    imageData: string
}